clang/test/Sema/constant-builtins-fmaximum-num.cpp

```diff
@@ -0,0 +1,55 @@
+// RUN: %clang_cc1 -std=c++17 -fsyntax-only -verify %s
+// FIXME: %clang_cc1 -std=c++17 -fsyntax-only -verify -fexperimental-new-constant-interpreter %s
+// expected-no-diagnostics
+
+constexpr double NaN = __builtin_nan("");
+constexpr double Inf = __builtin_inf();
+constexpr double NegInf = -__builtin_inf();
```

Comment:
Maybe could test snan cases, not sure if there are special rules for them in constant expressions

Reply:
It seems that we have no `__builtin_snan`. Do you have any idea how to produce an signaling nan?

Reply:
It's called __builtin_nans/__builtin_nansf